lib/jekyll/regenerator.rb

```diff
@@ -104,7 +104,10 @@ def modified?(path)
     def add_dependency(path, dependency)
       return if (metadata[path].nil? || @disabled)
 
-      metadata[path]["deps"] << dependency unless metadata[path]["deps"].include? dependency
+      if !metadata[path]["deps"].include? dependency
```

Comment:
A set sounds good. Wouldn't it actually speed it up, since it's constant time find (instead of linear time)?